File: Difficulty: Basic/Decimal to binary/decimal-to-binary.js
//{ Driver Code Starts
//Initial Template for javascript


'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}


function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let N = parseInt(readLine());
        let obj = new Solution();
        obj.toBinary(N);
    
console.log("~");
}
}

// } Driver Code Ends

// User function Template for javascript

class Solution {
    // Function to convert a number to its binary representation
    toBinary(N){
        var t = N, r = 0, arr = [], k = 1, p = 0;
        
        // While the number is not zero
        while (t !== 0) {
            // Get the remainder when dividing by 2 (this gives us the binary digits)
            r = t % 2;
            // Push the binary digit into the array
            arr.push(r);
            // Update the binary number in decimal form as we build it (p holds the result)
            p = p + r * k;
            // Multiply by 10 to shift the place value (moving to the next digit)
            k = k * 10;
            // Divide the number by 2 and take the floor to get the next bit
            t = Math.floor(t / 2);
        }

        // Print the result as the binary number in decimal form
        console.log(p)
    }
}

File: Difficulty: Easy/Decimal to binary/decimal-to-binary.c
//{ Driver Code Starts
//Initial Template for C

#include <stdio.h>


// } Driver Code Ends

#include <math.h>
// User function Template for C

void toBinary(int N)
{
    // Initialize the variable ans to store the binary result
    int ans = 0;
    int i = 0;  // Initialize the power index
    
    // Loop until N becomes 1
    while (N != 1)
    {
        // Add the current bit to the answer
        ans = ans + (N % 2) * pow(10, i);
        
        // Divide N by 2 to get the next bit
        N /= 2;
        
        // Increment the power index
        i++;
    }
    
    // Add the last bit (since N will be 1)
    ans = ans + (N % 2) * pow(10, i);
    
    // Print the final binary number
    printf("%d", ans);
}



//{ Driver Code Starts.

int main() {
	
	int t;
	scanf("%d", &t);
	
	
	while(t--)
	{
	    int n;
	    scanf("%d", &n);
	    
	    toBinary(n);
	    printf("\n");
	
printf("~%s", "\n");}
	return 0;
}
// } Driver Code Ends